src/services/reallocateCourtCaseToForce.ts

```diff
@@ -8,80 +8,110 @@ import insertNotes from "./insertNotes"
-    "SERIALIZABLE",
-    async (entityManager): Promise<UpdateResult | Error> => {
-      const courtCaseRepository = entityManager.getRepository(CourtCase)
-      let query = courtCaseRepository.createQueryBuilder().update(CourtCase)
-      const newForceCode = `${forceCode}${DEFAULT_STATION_CODE}`
-      query.set({ orgForPoliceFilter: newForceCode })
-      query = courtCasesByOrganisationUnitQuery(query, user) as UpdateQueryBuilder<CourtCase>
-      query.andWhere("error_id = :id", { id: courtCaseId })
-
-      const amendResult = await amendCourtCase(entityManager, { forceOwner: forceCode }, courtCaseId, user)
-
-      if (isError(amendResult)) {
-        throw amendResult
+  // - Reset triggers on reallocate
+  await dataSource.transaction("SERIALIZABLE", async (entityManager): Promise<void> => {
+    const events: AuditLogEvent[] = []
+
+    const courtCase = await getCourtCase(entityManager, courtCaseId)
+
+    if (isError(courtCase)) {
+      throw courtCase
+    }
+
+    if (!courtCase) {
+      throw new Error("Failed to unlock: Case not found")
+    }
+
+    const courtCaseRepository = entityManager.getRepository(CourtCase)
+    let query = courtCaseRepository.createQueryBuilder().update(CourtCase)
+    const newForceCode = `${forceCode}${DEFAULT_STATION_CODE}`
+    query.set({ orgForPoliceFilter: newForceCode })
+    query = courtCasesByOrganisationUnitQuery(query, user) as UpdateQueryBuilder<CourtCase>
+    query.andWhere("error_id = :id", { id: courtCaseId })
+
+    const amendResult = await amendCourtCase(entityManager, { forceOwner: forceCode }, courtCaseId, user)
+
+    if (isError(amendResult)) {
+      throw amendResult
+    }
+
+    const addNoteResult = await insertNotes(entityManager, [
+      {
+        noteText: `${user.username}: Case reallocated to new force owner: ${newForceCode}00`,
+        errorId: courtCaseId,
+        userId: "System"
       }
+    ])
 
-      const addNoteResult = await insertNotes(entityManager, [
+    if (isError(addNoteResult)) {
+      throw addNoteResult
+    }
+
+    if (note) {
+      const addUserNoteResult = await insertNotes(entityManager, [
         {
-          noteText: `${user.username}: Case reallocated to new force owner: ${newForceCode}00`,
+          noteText: note,
           errorId: courtCaseId,
-          userId: "System"
+          userId: user.username
         }
       ])
 
-      if (isError(addNoteResult)) {
-        throw addNoteResult
+      if (isError(addUserNoteResult)) {
+        throw addUserNoteResult
       }
+    }
 
-      if (note) {
-        const addUserNoteResult = await insertNotes(entityManager, [
-          {
-            noteText: note,
-            errorId: courtCaseId,
-            userId: user.username
-          }
-        ])
-
-        if (isError(addUserNoteResult)) {
-          throw addUserNoteResult
-        }
-      }
+    events.push(
+      getAuditLogEvent("information", "Hearing outcome reallocated by user", "Bichard New UI", {
```

Comment:
suggestion: I think it would be good to have the event category, event type and event source be defined in a type or enum somewhere, which would help prevent us from making typos in event codes and let us update the codes/source easily. I don't know if this is something that's on your radar for the refactor PR you've mentioned

Reply:
Thanks Alice! Yes, this is something that we thought about and we think it should exist in the core repository so we can then use it in the UI repo. We will open separate PRs for these improvements.